LunaDll/LuaMain/LuaProxyFFI.cpp

```diff
@@ -568,6 +569,79 @@ typedef struct ExtendedBlockFields_\
         std::wstring wpath = Str2WStr(path);
         return gCachedFileMetadata.exists(wpath);
     }
+
+    FFI_EXPORT(void) LunaLuaSetWindowTitle(const char* newName)
+    {
+        // Add "Software Rendered" if using it, like the normal window name code does
+        std::string fullName = newName;
+        if (gStartupSettings.softwareGL)
+        {
+            fullName += " (Software Renderer)";
+        }
+
+        // Convert the string to the required format
+        // From: https://stackoverflow.com/questions/27220/how-to-convert-stdstring-to-lpcwstr-in-c-unicode
+        // (Not compatible with UTF8? But not sure how easy that'd be to fix or if it's very relevant.)
+        std::wstring stemp = std::wstring(fullName.begin(), fullName.end());
+        LPCWSTR converted = stemp.c_str();
+
```

Comment:
You can use Str2WStr() call that already exists here (you'll find it at MiscFunctions). It uses proper way to convert UTF8 into UTF16

Reply:
I believe I tried that, but since fullName is a std::string rather than char, it didn't work. Might've just messed it up, though.

Reply:
If you want to pass the std::string/std::wstring into C call as char pointer, use the .c_str() call:

```cpp
SetWindowTextW(hwin, blabla.c_str());
```

Reply:
In the case of std::string, the char * pointer gets returned. And std::wstring returns wchar_t*.

Reply:
Specifically in this case it should just be a matter of
```c++
SetWindowTextW(gMainWindowHwnd, Str2WStr(fullName).c_str());
```

Reply:
Unfortunately, this acts the same as the current code (though more concise, which is nice), in that it just displays question marks.

`Misc.setWindowTitle("The Chucksode これが何を意味するのかわかりません")`

![image](https://user-images.githubusercontent.com/59939907/162585157-1ca8010e-39ad-44f6-8a42-bc9cf78d0cd3.png)

Reply:
Did you save your lua file as UTF8?

Reply:
Should be the case, as it's VSC's default.
![image](https://user-images.githubusercontent.com/59939907/162585242-5532b28a-5523-4881-974b-9cbe2e7958bc.png)

Reply:
I think unicode is getting to `SetWindowTextW` just fine. Maybe I'm wrong, but I'm guessing that [IsWindowUnicode()](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-iswindowunicode) probably returns `FALSE` for the main window... in which case Windows may behind the scenes be converting the unicode passed to `SetWindowTextW` back to ANSI.

This may be a little bit of a hack, and it may have other consequences, but [this stackoverflow answer](https://stackoverflow.com/questions/324624/multi-byte-character-set-in-mfc-application/338634#338634) I came across makes me suspect that the following snippet may have some impact if done before the `SetWindowTitleW` (though may just as easily crash)
`SetWindowLongW( gMainWindowHwnd, GWL_WNDPROC, GetWindowLongA( gMainWindowHwnd, GWL_WNDPROC ) );`

Reply:
Please commit your current changes to your branch to let me review, I could help you once I see your code that gives this result

Reply:
Done.